ng-dev/pr/merge/target-label.ts

```diff
@@ -27,19 +63,19 @@ export class InvalidTargetLabelError {
 }
 
 /** Gets the target label from the specified pull request labels. */
-export async function getTargetLabelFromPullRequest(
+export async function getMatchingTargetLabelForPullRequest(
   config: Pick<MergeConfig, 'noTargetLabeling'>,
-  labels: string[],
+  labelsOnPullRequest: string[],
+  allTargetLabels: TargetLabel[],
 ): Promise<TargetLabel> {
   if (config.noTargetLabeling) {
     throw Error('This repository does not use target labels');
   }
 
-  const targetLabels = await getTargetLabels();
-  /** List of discovered target labels for the PR. */
-  const matches = [];
-  for (const label of labels) {
-    const match = targetLabels.find(({pattern}) => matchesPattern(label, pattern));
+  const matches: TargetLabel[] = [];
+
+  for (const label of labelsOnPullRequest) {
+    const match = allTargetLabels.find(({name}) => label === name);
```

Comment:
```suggestion
   if (allTargetLabels.includes(label)) {
```

Reply:
If I don't miss anything, I think we need a reference to the marched target label.

Reply:
Whoops, your right.